homeassistant/components/sensor/filter.py

```diff
@@ -403,19 +406,23 @@ def __init__(self, window_size, precision, entity, radius):
         self._radius = radius
         self._stats_internal = Counter()
 
+    def filter_state(self, new_state, store_raw=True):
+        return super().filter_state(new_state, store_raw)
+
     def _filter_state(self, new_state):
         """Implement the outlier filter."""
+        median = statistics.median([s.state for s in self.states]) \
+            if len(self.states) > 0 else 0
         if (len(self.states) == self.states.maxlen and
-                abs(new_state.state -
-                    statistics.median([s.state for s in self.states])) >
+                abs(new_state.state - median) >
                 self._radius):
 
             self._stats_internal['erasures'] += 1
 
             _LOGGER.debug("Outlier nr. %s in %s: %s",
                           self._stats_internal['erasures'],
                           self._entity, new_state)
-            return self.states[-1]
```

Comment:
return median

Reply:
median is a bare float at this point though, not a FilterState, and the caller expects `.state` and `.timestamp` properties on it.
 
The conditional falls through anyway and calls `return new_state` below after updating the `.state` property as written - I could alternatively pull the last value out of `self.states[]`, update its `.state` and return that if you think that's clearer.

Reply:
One issue with updating the last `self.states` entry would be that the timestamp never updates when outlier is taking effect, so the "new" event's timestamp is never written to the history (just a duplicate of the previous timestamp). To me, it makes sense that outlier returns the **new** state's timestamp, except with its value swapped for the filtered one so that the historical timeline makes sense (instead of having a gap)

Reply:
OK lets leave it with `new_state.state = median`, for me it was just a matter of explicitly terminating the branch with a return, but this is fine.